src/plugins/intel_gna/src/frontend/scale_factor_calc.cpp

```diff
@@ -306,8 +306,8 @@ bool ScaleFactorCalculator::requantizeInput(InferenceEngine::CNNLayerPtr input,
         return false;
     }
 
-    auto prevLayer = InferenceEngine::CNNNetHasPrevLayer(layer.get(), prevInputIdx)
-                         ? InferenceEngine::CNNNetPrevLayer(layer, prevInputIdx)
+    auto prevLayer = InferenceEngine::CNNNetHasPrevLayer(layer.get(), static_cast<int>(prevInputIdx))
```

Comment:
you set second argument to size_t and not you cast to int? 
inline bool CNNNetHasPrevLayer(const InferenceEngine::CNNLayer* layer, size_t idx = 0) {

shouldn't this produce warning?

https://github.com/openvinotoolkit/openvino/pull/17841/files#diff-637b2179f395ecedb9fb41cb1ee70aa666b3b7c2ce3d8fb435b9c3c8bbc248ffR355